pkg/utils/utils.go

```diff
@@ -44,3 +48,92 @@ func ContextWithDeadlineFn(ctx context.Context, deadlineFn func(orig time.Time)
 	}
 	return ctx, cancel
 }
+
+func FitsInNBitsSigned(n int, bi *big.Int) bool {
+	if bi.Sign() < 0 {
+		bi = new(big.Int).Neg(bi)
+		bi.Sub(bi, big.NewInt(1))
+	}
+	return bi.BitLen() <= n-1
+}
+
+func MergeValueFields(valueFields []map[string]any) (map[string]any, error) {
```

Comment:
This function and the one below will be used to create a MapDecoder that can decode to maps.  That can be re-used so that implementations don't need to decode to interfaces directly.  The need for this comes from encoding for functions (or similar) where instead of an object representing fields, you get an array of objects squashed into one object and then encoded to the fields on that object.